compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/descriptors/JvmDeclarationFactory.kt

```diff
@@ -152,6 +155,36 @@ class JvmDeclarationFactory(
         else
             getFieldForObjectInstance(singleton)
 
+    fun getStaticBackingField(irProperty: IrProperty): IrField? {
+        // Only fields defined directly in objects should be made static.
+        // Fake overrides never point to those, as objects are final.
+        if (irProperty.origin == IrDeclarationOrigin.FAKE_OVERRIDE) return null
+        val oldField = irProperty.backingField ?: return null
+        val oldParent = irProperty.parent as? IrClass ?: return null
+        if (!oldParent.isCompanion && !oldParent.isObject) return null
```

Comment:
Looks like `!oldParent.isObject` here is unnecessary

Reply:
The condition is the other way around, so it's actually `!oldParent.isCompanion` that's unnecessary.